src/coreclr/tools/aot/crossgen2/crossgen2.csproj

```diff
@@ -1,14 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
+    <CrossHostArch Condition="'$(CrossBuild)' == 'true' or '$(TargetArchitecture)' != '$(BuildArchitecture)' or '$(HostOS)' != '$(TargetOS)' or '$(EnableNativeSanitizers)' != ''">$(BuildArchitecture)</CrossHostArch>
```

Comment:
For ilc, this is set unconditionally in the crossarch project
https://github.com/dotnet/runtime/blob/2267228c5115633a8ce310e4a812678a45a170db/src/coreclr/tools/aot/ILCompiler/ILCompiler_crossarch.csproj#L4

Also, we should make the crossgen and ilc project files use the same naming scheme. I like the naming scheme for ilc project files (ILCompiler.csproj, ILCompiler_crossarch.csproj).